docs/guides/debugging.md

```diff
@@ -0,0 +1,91 @@
+# Debugging
+
+## Debugging? What it is and why?
+
+Debugging is one of **the most important things on C**, not only Concord. It is important **because it shows where the error** (AKA segmentation fault or core dump) **came from** since you cannot know it with ease without it.
+
+We will be focusing on two popular tools for debugging, `GDB` and `Valgrind`.
+
+You could say "`But doesn't the compiler already tell the errors?`". Well, it does, but it is not perfect, and things like WebSocket messages cannot predict and say if it's going to work fine, or it's going to fail, don't trust blindly it.
+
+## Debugging (segmentation faults and core dumps)
+
+For this guide, we are going to use GDB, an easy debugging tool for those types of errors.
+
+### Installing GDB
+
+Before using it, we must install it. We can easily do this with apt, but first, let's check if your machine already has it:
+
+```console
+$ gdb
+```
+
+In case this command failed, we can proceed with the installation `GDB` (expected to be running apt with root permissions):
+
+```console
+# apt install gdb
+```
+
+### Notes before debugging
+
+For better debugging results, **you must disable all types of compiler optimizations** (-O1, -O2, -O3, -Ofast and etc) and **enable the `-g` flag**.
+
+This is important, enabling **optimization flags will just make debugging harder** and maybe you will be not enabled to see the backtrace with it.
+
+### Proceeding with debugging
+
+Since we now installed `GDB`, we can use it. For example, if the file that you want to debug is named `ConcordGuides`, the command that you will need to execute is:
+
+```console
+$ gdb ./ConcordGuides
+``` 
+
+After we executed this command, you should see some Copyright messages, just press `c` and `enter`. and you are now inside GDB's shell.
+
+To start debugging, just **use the command `r`** (shorter version of `run`) and GDB **will start running and debugging your program**. You will **need to wait until your program/bot crashes** to see the results.
+
+After your program/bot crashes, you can run the command `bt` to **see the backtrace of the crash**, you will need to find where is the line of your code that is throwing the segmentation fault/core dumped (not always the crash comes from your code, maybe it's a bug from a library, but here we are expecting to be from your code) and see what's wrong with it.
```

Comment:
"After your program/bot crashes, you can run the command `bt` to **see the backtrace of the crash**. You will need to find which line of your code is throwing the segmentation fault/coredump (please note that the crash doesn't always come from your program, it could be from a library), and see what's wrong with it."